MapboxStatic/Snapshot.swift

```diff
@@ -503,49 +503,52 @@ public class Snapshot: NSObject {
             let apiMessage = json["message"] as? String
             guard image != nil && error == nil && apiMessage == nil else {
                 let apiError = Snapshot.descriptiveError(json, response: response, underlyingError: error)
-                dispatch_async(dispatch_get_main_queue()) {
-                    handler(image: nil, error: apiError)
+                DispatchQueue.main.async {
+                    handler(nil, apiError)
                 }
                 return
             }
             
-            dispatch_async(dispatch_get_main_queue()) {
-                handler(image: image, error: nil)
+            DispatchQueue.main.async {
+                handler(image, nil)
             }
-        }
+        }) 
         task.resume()
         return task
     }
     
     /**
      Returns an error that supplements the given underlying error with additional information from the an HTTP response’s body or headers.
      */
-    private static func descriptiveError(json: JSONDictionary, response: NSURLResponse?, underlyingError error: NSError?) -> NSError {
-        var userInfo = error?.userInfo ?? [:]
-        if let response = response as? NSHTTPURLResponse {
+    fileprivate static func descriptiveError(_ json: JSONDictionary, response: URLResponse?, underlyingError error: Error?) -> NSError {
+        var userInfo = [String: String]()
+        var statusCode:Int = -1
```

Comment:
This variable isn’t used outside of the `if let` block below, so it doesn’t need to be declared up here.